packages/pnpm/src/main.ts

```diff
@@ -134,6 +134,15 @@ export default async function run (inputArgv: string[]) {
     }
   }
 
+  if (
+    cmd === 'install' &&
+    typeof workspaceDir === 'string' &&
+    config.recursiveInstall
+  ) {
+    cliOptions['recursive'] = true
+    config.recursive = true
```

Comment:
because we pass cliOptions to `getConfig` as default, when cliOption.recursive is true, we also need to set config.recursive to tru too